### PR TITLE
Stock locations - Hierarchical view for better visual representation

### PR DESCRIPTION
**Pull request by jat-odoo**
Description of the issue/feature this PR addresses:
* Task: https://www.odoo.com/web#id=1873106&action=327&model=project.task&view_type=form&menu_id=4720
* Pad: https://pad.odoo.com/p/r.86fbd89e4b02e1e80aa847474c13678d

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/static/src/scss/stock_location_hierarchy.scss [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+.o_stock_location_hierarchy {
+    .o_hierarchy_container {
+        border: 1px solid gray('300');
+        background-color: lighten(gray('200'), 2%);
+        border-radius: .5rem;
+        white-space: nowrap;
+        overflow-x: auto;
+    }
+    ul {
```
Comment (sle-odoo) [inline addons/stock/static/src/scss/stock_location_hierarchy.scss]: @jat-odoo can you add a class to the ul on the template and make this selector more specific? ie ul.o_hierarchy_something


same for li

File: addons/stock/models/stock_location.py [hunk 2]
```diff
@@ -163,6 +163,33 @@ def should_bypass_reservation(self):
...
+        for i in result:
+            i['childs'] = []
+            data[i['id']] = i
+
+        for i in data:
+            location_id = data[i]['location_id']
+            if location_id and data.get(location_id[0]):
+                data[location_id[0]]['childs'].append(data[i])
+            else:
+                roots.append(data[i])
```
[the remark this answers is not in the document]
Comment (dja-odoo) [reply addons/stock/models/stock_location.py]: Done

File: addons/stock/models/stock_location.py [hunk 3]
```diff
@@ -163,6 +163,33 @@ def should_bypass_reservation(self):
...
+
+        values['data'] = roots
+        return values
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu=False):
+        result = super().fields_view_get(view_id, view_type, toolbar, submenu)
+        if result['type'] == 'qweb':
```
[the remark this answers is not in the document]
Comment (dja-odoo) [reply addons/stock/models/stock_location.py]: Done

File: addons/stock/static/src/scss/stock_location_hierarchy.scss (hunk 1 above)
Comment (dja-odoo) [reply addons/stock/static/src/scss/stock_location_hierarchy.scss]: Done

File: addons/web/static/src/js/views/qweb/qweb_view.js [hunk 4]
```diff
@@ -66,8 +66,7 @@ var Model = AbstractModel.extend({
      * reload
      */
     reload: function (_id, params) {
-        _.extend(this._state, _.pick(params, ['domain', 'context']));
-
+        _.extend(this._state, _.pick(params, ['viewId', 'modelName', 'domain', 'context']));
```
[the remark this answers is not in the document]
Comment (dja-odoo) [reply addons/web/static/src/js/views/qweb/qweb_view.js]: @amoyaux I couldn't find any usecase either. I think we can get rid of it.

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**